src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -77,41 +91,295 @@
         }
-        nuiManager.closeScreen("GooKeeper:gooeyActivateScreen");
+        
+        gooeyEntity.addOrSaveComponent(matingComponent);
+
+        if (matingComponent.selectedForMating && matingComponent.matingWithEntity != EntityRef.NULL) {
+            gooeyEntity.send(new BeginBreedingEvent(event.getInstigator(), gooeyEntity, matingComponent.matingWithEntity));
+        }
+    }
+
+    /**
+     * Receives the BeginBreedingEvent when the actual breeding process is to be initiated.
+     *
+     * @param event
+     * @param gooeyEntity
+     * @param gooeyComponent
+     */
+    @ReceiveEvent
+    public void onBeginBreedingProcess(BeginBreedingEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent) {
+        delayManager.addDelayedAction(gooeyEntity, Constants.spawnGooeyEggEventID, random.nextLong(4000, 7000));
+    }
+
+    /**
+     * Receives the AfterGooeyBreedingEvent when the breeding process is to be ended
+     *
+     * @param event
+     * @param gooeyEntity
+     * @param gooeyComponent
+     * @param matingComponent
+     */
+    @ReceiveEvent
+    public void onBreedingProcessEnd(AfterGooeyBreedingEvent event, EntityRef gooeyEntity, GooeyComponent gooeyComponent, MatingComponent matingComponent) {
+        EntityRef offspringEntity = event.getOffspringGooey();
+        SkeletalMeshComponent skeletalMeshComponent = gooeyEntity.getComponent(SkeletalMeshComponent.class);
+        skeletalMeshComponent.material = getOffspringMaterial(gooeyEntity, matingComponent.matingWithEntity, offspringEntity);
+        skeletalMeshComponent.mesh = null;
+        offspringEntity.addComponent(skeletalMeshComponent);
+
+        MatingComponent matingComponent1 = matingComponent.matingWithEntity.getComponent(MatingComponent.class);
+
+        matingComponent.selectedForMating = false;
+        matingComponent1.selectedForMating = false;
+
+        CharacterMovementComponent characterMovementComponent = gooeyEntity.getComponent(CharacterMovementComponent.class);
+        CharacterMovementComponent characterMovementComponent1 = matingComponent.matingWithEntity.getComponent(CharacterMovementComponent.class);
+
+        if (characterMovementComponent != null) {
+            characterMovementComponent.speedMultiplier = 1f;
+        }
+
+        if (characterMovementComponent1 != null) {
+            characterMovementComponent1.speedMultiplier = 1f;
+        }
+
+        gooeyEntity.saveComponent(characterMovementComponent);
+        matingComponent.matingWithEntity.saveComponent(characterMovementComponent1);
+    }
+
+    /**
+     * Receives DelayedActionTriggeredEvent, which spawns the gooey egg
+     *
+     * @param event     The DelayedActionTriggeredEvent event
+     * @param gooeyEntity    The entity to which the event is sent
+     */
+    @ReceiveEvent
+    public void onDelayedAction(DelayedActionTriggeredEvent event, EntityRef gooeyEntity) {
+        if (event.getActionId().equals(Constants.spawnGooeyEggEventID)) {
```

Comment:
Likewise to above, both this and the `else` section should be moved to separate methods